proposals/lip-0052.md

```diff
@@ -362,712 +379,1832 @@ The module provides the following commands to modify the NFT store.
+    recipientAddress: Address,
+    nftID: NFTID
+) -> None:
+
+    delete entry userStore(senderAddress, nftID) from the user substore
+    createUserEntry(recipientAddress, nftID)
+    NFTStore(nftID).owner = recipientAddress
+
+    emitEvent(
+        module = MODULE_NAME_NFT,
+        name = EVENT_NAME_TRANSFER,
+        data = {
+            "senderAddress": senderAddress,
+            "recipientAddress": recipientAddress,
+            "nftID": nftID,
+            "result": RESULT_SUCCESSFUL
+        },
+        topics = [senderAddress, recipientAddress]
+    )
+```
+
+#### transferCrossChainInternal
+
+```python
+def transferCrossChainInternal(
+    senderAddress: Address,
+    recipientAddress: Address,
+    nftID: NFTID,
+    receivingChainID: ChainID,
+    messageFee: uint64,
+    data: str
+) -> None:
+
+    delete entry userStore(senderAddress, nftID) from the user substore
+
+    if getChainID(nftID) == OWN_CHAIN_ID:
+        NFTStore(nftID).owner = receivingChainID
+
+    if getChainID(nftID) == receivingChainID:
+        delete entry NFTStore(nftID) from the NFT substore
+
+    emitEvent(
+        module = MODULE_NAME_NFT,
+        name = EVENT_NAME_TRANSFER_CROSS_CHAIN,
+        data = {
+            "senderAddress": senderAddress,
+            "recipientAddress": recipientAddress,
+            "nftID": nftID,
+            "receivingChainID": receivingChainID,
+            "result": RESULT_SUCCESSFUL
+        },
+        topics = [senderAddress, recipientAddress, receivingChainID]
+    )
+
+    Interoperability.send(
+        sendingAddress = senderAddress,
+        module = MODULE_NAME_NFT,
+        crossChainCommand = CROSS_CHAIN_COMMAND_NAME_TRANSFER,
+        receivingChainID = receivingChainID,
+        fee = messageFee,
+        params = {
+            "nftID": nftID,
+            "senderAddress": senderAddress,
+            "recipientAddress": recipientAddress,
+            "data": data
+        }
+    )
+```
+
+### Protocol Logic for Other Modules
+
+#### isNFTSupported
+
+This function returns the support status of an NFT.
+
+```python
+def isNFTSupported(nftID: NFTID) -> bool:
+
+    if NFTStore(nftID) does not exist:
+        raise Exception("NFT substore entry does not exist.")
+
+    chainID = getChainID(nftID)
+
+    if getChainID(nftID) == OWN_CHAIN_ID:
+        return True
+
+    if supportedNFTsStore(ALL_SUPPORTED_NFTS_KEY) exists:
+        return True
+
+    if supportedNFTsStore(chainID) exists:
+        if supportedNFTsStore(chainID).supportedNftIDArray == []:
+            return True
+        if nftID is in supportedNFTsStore(chainID).supportedNftIDArray:
+            return True
+
+    return False
+```
+
+#### getChainID
+
+This function returns the native chain of an NFT.
+
+```python
+def getChainID(nftID: NFTID) -> ChainID:
+
+    if NFTStore(nftID) does not exist:
+        raise Exception("NFT substore entry does not exist.")
+
+    return nftID[:CHAIN_ID_LENGTH]
+```
+
+#### getAttributes
```

Comment:
I would rename this `getAttributesArray` and have another function to just get the attributes for a certain module `getAttributes(nftID: NFTID, module: str) -> bytes`

Reply:
Since you are returning, you can just raise at the end and remove the `attributes_exist` variable all together